### PR TITLE
Add ESLint for JavaScript linting

### PR DESCRIPTION
**Pull request by matthewroach**
Adding ESlint and the dependent packages based on configuration

Added two new `npm` scripts

* `npm run lint` - to run ESLint
* `npm run lint:fix` - to use the ESLint fix CLI option for it to automatically fix issues it's able to

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	rules: {
+		'global-require': 'off',
+		indent: ['error', 'tab'],
+		'no-tabs': ['error', { allowIndentationTabs: true }],
+		'react/jsx-indent': [2, 'tab'],
+		'react/jsx-indent-props': [2, 'tab'],
+		'no-underscore-dangle': ['error', { allow: ['_website', '_id'] }],
+		'import/no-extraneous-dependencies': 'off', // This might be fine. It's worth looking into at the very least.
+		'import/no-unresolved': [2, {
+			ignore: ['react', '^fusion:.+$', '^@arc-test-org/.+$', '^@arc-publishing/.+$', '^@wpmedia/.+$', '~/blocks.json'],
```
Comment (JackHowa) [inline .eslintrc.js]: don't think we should need these fusion-specific ignores for this repo, but it's just a suggestion

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		'jsx-a11y/no-noninteractive-element-to-interactive-role': 'off',
+		'jsx-a11y/no-noninteractive-tabindex': 2,
+		'jsx-a11y/no-onchange': 2,
+		'jsx-a11y/no-redundant-roles': 2,
+		'jsx-a11y/no-static-element-interactions': 2,
+		'jsx-a11y/role-has-required-aria-props': 2,
+		'jsx-a11y/role-supports-aria-props': 2,
+		'jsx-a11y/scope': 2,
+		'jsx-a11y/tabindex-no-positive': 2,
+		'react-hooks/rules-of-hooks': 'error',
```
Comment (JackHowa) [inline .eslintrc.js:80]: good good

File: src/components/paragraph/index.jsx [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import React from 'react';
```
Comment (JackHowa) [inline src/components/paragraph/index.jsx]: don't think we technically need this with react 17 but no big deal https://stackoverflow.com/a/64646593/7491536

**Review by JackHowa — APPROVED** [review]
Works as expected. Only minor suggestions:

before lint:fix

Screen Shot 2022-01-21 at 13 05 36
after lint fix 

Screen Shot 2022-01-21 at 13 05 46

lint identifies console log
Screen Shot 2022-01-21 at 13 05 22

File: src/components/paragraph/index.jsx (hunk 3 above)
Comment (matthewroach) [reply src/components/paragraph/index.jsx]: ESlint throws an error if it's not there

**Review by JackHowa — APPROVED** [review]
looks great! thanks 
Screen Shot 2022-01-21 at 13 27 14

**Pull request merged**